packages/datadog-plugin-cypress/src/plugin.js

```diff
@@ -248,19 +250,23 @@ module.exports = (on, config) => {
     const cypressTests = tests || []
     const finishedTests = finishedTestsByFile[spec.relative] || []
 
-    // Get tests that didn't go through `dd:afterEach` and haven't been skipped by ITR
+    // Get tests that didn't go through `dd:afterEach`
     // and create a skipped test span for each of them
     cypressTests.filter(({ title }) => {
       const cypressTestName = title.join(' ')
-      const isSkippedByItr = testsToSkip.find(test =>
-        cypressTestName === test.name && spec.relative === test.suite
-      )
       const isTestFinished = finishedTests.find(({ testName }) => cypressTestName === testName)
 
-      return !isSkippedByItr && !isTestFinished
+      return !isTestFinished
     }).forEach(({ title }) => {
-      const skippedTestSpan = getTestSpan(title.join(' '), spec.relative)
+      const cypressTestName = title.join(' ')
+      const isSkippedByItr = testsToSkip.find(test =>
```

Comment:
If I understood the code correctly, we're iterating over the `testsToSkip`, which seems to be an array, for every test case. So if we have N tests all of which are skipped, the time complexity will be O(N^2).
I don't know how many Cypress tests a typical project has, but potentially this could be a scalability bottleneck.
Perhaps there's a data structure that allows a more efficient lookup, such as a hash set?

Reply:
yeah these are tests that are slow by nature so I don't expect the number to be big (the maximum will be in the hundreds more than in the thousands). Good catch though! I'll make a note